src/Workspaces/Core/Portable/CodeActions/CodeAction.cs

```diff
@@ -284,57 +293,78 @@ public static CodeAction Create(string title, Func<CancellationToken, Task<Solut
             return new SolutionChangeAction(title, createChangedSolution, equivalenceKey);
         }
 
-        internal class DocumentChangeAction : CodeAction
+        /// <summary>
+        /// Creates a top level code action with multiple code action choices a user can invoke.
+        /// </summary>
+        public static CodeAction Create(string title, IEnumerable<CodeAction> nestedActions, string equivalenceKey = null)
+        {
+            if (title == null)
+            {
+                throw new ArgumentNullException(nameof(title));
+            }
+
+            return new SimpleCodeAction(title, nestedActions.AsImmutableOrEmpty(), equivalenceKey);
+        }
+
+        internal class SimpleCodeAction : CodeAction
         {
             private readonly string _title;
-            private readonly Func<CancellationToken, Task<Document>> _createChangedDocument;
             private readonly string _equivalenceKey;
+            private readonly ImmutableArray<CodeAction> _nestedActions;
 
-            public DocumentChangeAction(string title, Func<CancellationToken, Task<Document>> createChangedDocument, string equivalenceKey = null)
+            public SimpleCodeAction(string title, ImmutableArray<CodeAction> nestedActions, string equivalenceKey)
             {
                 _title = title;
-                _createChangedDocument = createChangedDocument;
+                _nestedActions = nestedActions;
                 _equivalenceKey = equivalenceKey;
             }
 
-            public override string Title
+            public sealed override string Title => _title;
+            public sealed override string EquivalenceKey => _equivalenceKey;
+
+            internal override bool IsInvokable => false;
```

Comment:
what is this Property for? Its `true` in the abstract base class, `false` here and `true` in all the derived classes. I guess I couldn't understand. Could you please explain. Alternatively doc comments would be nice too.